packages/@aws-cdk/aws-iam/lib/managed-policy.ts

```diff
@@ -316,3 +321,30 @@ export class ManagedPolicy extends Resource implements IManagedPolicy {
     return result;
   }
 }
+
+/**
+ * A `grantPrincipal` of ManagedPolicy.
+ */
+class ManagedPolicyGrantPrincipal implements IPrincipal {
+  public readonly assumeRoleAction = 'sts:AssumeRole';
+  public readonly grantPrincipal: IPrincipal;
+  public readonly principalAccount?: string;
+
+  constructor(private _managedPolicy: ManagedPolicy) {
+    this.grantPrincipal = this;
+    this.principalAccount = _managedPolicy.env.account;
+  }
+
+  public get policyFragment(): PrincipalPolicyFragment {
+    throw new Error(`Cannot use a ManagedPolicy ${this._managedPolicy.node.path} as the 'Principal' or 'NotPrincipal' in an IAM Policy`);
```

Comment:
```suggestion
    throw new Error(`Cannot use a ManagedPolicy '${this._managedPolicy.node.path}' as the 'Principal' or 'NotPrincipal' in an IAM Policy`);
```

Can you add a comment above this error explaining why we need to throw it? It's not immediately clear

Reply:
I believe `policyFragment` is currently the best location to raise an error when we try to add Policy/ManagedPolicy to a resource-based policy. Suggestions are welcome.

Reply:
@Tietew I know this is an old thread but I have a question wanting to understanding better.

You mentioned that `PolicyStatement.validatePolicyPrincipal()` throws error for `Group`. So why not update `PolicyStatement.validatePolicyPrincipal()` to throw for Policy or Managed Policy as well?